### PR TITLE
mds: add set logicalpool status

### PR DESCRIPTION
**Pull request by xu-chaojie**
### What problem does this PR solve?

add allocate status when create logicalpool and add SetLogicalPool to tools

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [x] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: src/mds/topology/topology_item.h [hunk 1]
```diff
@@ -103,7 +103,7 @@ class LogicalPool {
           type_(PAGEFILE),
           initialScatterWidth_(0),
           createTime_(0),
-          status_(AllocateStatus::DENY),
+          status_(AllocateStatus::ALLOW),
```
Comment (wu-hanqing) [inline src/mds/topology/topology_item.h:106]: 这里默认为什么调整为Allow？

File: proto/topology.proto [hunk 2]
```diff
@@ -345,6 +344,7 @@ message CreateLogicalPoolRequest {
     required bytes redundanceAndPlaceMentPolicy = 5;  //json body
     required bytes userPolicy = 6;                    //json body
     optional uint32 scatterWidth = 7;      //生成copyset依据的scatterWidth平均值
+    optional AllocateStatus status = 8;
```
Comment (ilixiaocui) [inline proto/topology.proto:347]: commit message中以涉及到修改的模块开头
mds: ****

File: proto/topology.proto [hunk 3]
```diff
@@ -387,6 +387,14 @@ message ListLogicalPoolResponse {
     repeated LogicalPoolInfo logicalPoolInfos = 2;
 }
 
+message SetLogicalPoolRequest {
+    optional uint32 logicalPoolID = 1;
```
Comment (ilixiaocui) [inline proto/topology.proto:391]: 这里为什么是optional

File: src/mds/topology/topology_item.h (hunk 1 above)
Comment (ilixiaocui) [reply src/mds/topology/topology_item.h:106]: 没有DENY的测试用例吗

File: tools/curvefsTool.cpp [hunk 4]
```diff
@@ -1086,6 +1117,51 @@ int CurvefsTools::SetChunkServer() {
     return 0;
 }
 
+int CurvefsTools::SetLogicalPool() {
```
Comment (ilixiaocui) [inline tools/curvefsTool.cpp:1120]: 这个没有测试用例吗

File: src/mds/topology/topology_item.h (hunk 1 above)
Comment (xu-chaojie) [reply src/mds/topology/topology_item.h:106]: 这是因为目前logicalpool在创建时就是默认改为Allow的，防止初始化之后变成deny就完蛋了

File: proto/topology.proto (hunk 3 above)
Comment (xu-chaojie) [reply proto/topology.proto:391]: done

File: tools/curvefsTool.cpp (hunk 4 above)
Comment (xu-chaojie) [reply tools/curvefsTool.cpp:1120]: 工具这块目前都没有测试

File: src/mds/topology/topology_item.h (hunk 1 above)
Comment (xu-chaojie) [reply src/mds/topology/topology_item.h:106]: > 没有DENY的测试用例吗

有，见 test/mds/topology/test_topology.cpp:1116,  UpdateLogicalPoolAllocateStatus_success 这个用例

File: proto/topology.proto (hunk 2 above)
Comment (xu-chaojie) [reply proto/topology.proto:347]: done

**Review by ilixiaocui — APPROVED** [review]

**Comment by xu-chaojie** [comment]
recheck

**Pull request merged**